### PR TITLE
Terraform Proposal

### PR DESCRIPTION
**Pull request by cyberbuff**
Executor Changes  to run terraform commands which are necessary for [Terraform](https://github.com/redcanaryco/atomic-red-team/pull/2185) in atomic-red-team. The executor will run `terraform apply` during GetPreReqs and `terraform destroy` during cleanup.

### CODE REVIEW

File: Public/Invoke-AtomicTest.ps1 [hunk 1]
```diff
@@ -405,6 +439,9 @@ function Invoke-AtomicTest {
                         Write-PrereqResults $FailureReasons $testId
                     }
                     elseif ($GetPrereqs) {
+                        if($(Test-IncludesTerraform $AT $testCount)){
+                            Build-TFVars $AT $testCount $InputArgs
```
Comment (cyberbuff) [inline Public/Invoke-AtomicTest.ps1:442]: Override terraform variables(s3 bucket name, etc) when input arguments are provided via powershell.

File: Public/Invoke-AtomicTest.ps1 [hunk 2]
```diff
@@ -439,6 +476,9 @@ function Invoke-AtomicTest {
                         $final_command = Merge-InputArgs $test.executor.cleanup_command $test $InputArgs $PathToPayloads
                         $res = Invoke-ExecuteCommand $final_command $test.executor.name $executionPlatform $TimeoutSeconds $session -Interactive:$Interactive
                         Write-KeyValue "Done executing cleanup for test: " $testId
+                        if($(Test-IncludesTerraform $AT $testCount)){
+                            Remove-TerraformFiles $AT $testCount
```
Comment (cyberbuff) [inline Public/Invoke-AtomicTest.ps1:479]: Remove terraform state files and overrided terraform variables if present.

**Comment by patel-bhavin** [comment]
This PR has been tested with this atomic: https://github.com/redcanaryco/atomic-red-team/pull/2185
 Imported atomic module from @cyberbuff's fork and tested.
 
image

**Review by patel-bhavin — APPROVED** [review]

**Comment by cyberbuff** [comment]
Powershell executors for Linux and MacOS have been fixed now. https://github.com/redcanaryco/invoke-atomicredteam/pull/115

**Comment by patel-bhavin** [comment]
@cyberbuff : Thank you so much for spending time on this new and slick feature and for your patience! I believe this introduces a really nice flavor of writing and thinking of attacks in cloud. 

We will do a more formal announcement from Atomic Red team side in the next few weeks!!!

**Review by patel-bhavin — APPROVED** [review]

**Pull request merged**